core/testsuite/src/test/java/com/blazebit/persistence/testsuite/DateDiffTest.java

```diff
@@ -398,11 +444,19 @@ public void testDateWithParameterDiff() {
         assertEquals(minutesBetween(c2, c1), actual.get(18));
         assertEquals(secondsBetween(c2, c1), actual.get(19));
         assertEquals(millisecondsBetween(c2, c1), actual.get(20));
+        assertEquals(secondsBetween(c2, c1), actual.get(21));
+        assertEquals(microsecondsBetween(c2, c1), actual.get(22));
+        assertEquals(weeksBetween(c2, c1), actual.get(23));
+        assertEquals(quartersBetween(c2, c1), actual.get(24));
     }
 
     public static long millisecondsBetween(Calendar day1, Calendar day2) {
         return day2.getTimeInMillis() - day1.getTimeInMillis();
     }
+
+    public static long microsecondsBetween(Calendar day1, Calendar day2) {
+        return (day2.getTimeInMillis() - day1.getTimeInMillis()) * 1000;
```

Comment:
Since the best precision we have with Calendar is milliseconds, we should add a test that uses DBMS literals/functions to produce values with microseconds which we can check against. Currently we don't know for sure if we are correctly extracting microsecond parts